src/queue.cpp

```diff
@@ -912,6 +957,41 @@ cvk_command_kernel::build_batchable_inner(cvk_command_buffer& command_buffer) {
         return CL_OUT_OF_RESOURCES;
     }
 
+    // Setup printf buffer descriptor if needed
+    if (m_kernel->program()->uses_printf()) {
+        // Create and initialize the printf buffer
+        auto buffer = m_queue->printf_buffer();
+        if (buffer->map()) {
+            memset(buffer->host_va(), 0, 4);
+            buffer->unmap();
+        }
```

Comment:
Maybe this could be encapsulated into a `reset_printf_buffer`?

Reply:
Done